cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.cc

```diff
@@ -0,0 +1,156 @@
+
+namespace h5vcc_experiments {
+
+namespace {
+
+mojom::OverrideState GetFeatureInternal(
+    std::map<std::string, h5vcc_experiments::mojom::OverrideState>&
+        override_map,
+    const std::string& feature_name) {
+  // auto accessor = cobalt::GlobalFeatures::GetInstance()->accessor();
+  // base::FeatureList::OverrideState override =
+  //     accessor->GetOverrideStateByFeatureName(
+  //         feature_name);
+  // if (override == base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE)
+  // {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+  // } else if (override ==
+  // base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE) {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_DISABLE_FEATURE;
+  // }
+  // return h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+
+  // Cache feature params in a map for faster lookup.
+  if (override_map.empty()) {
+    std::string enable_features;
+    std::string disable_features;
+
+    base::FeatureList::GetInstance()->GetFeatureOverrides(&enable_features,
+                                                          &disable_features);
+
+    std::string feature;
+    std::string study;
+    std::string group;
+    std::string feature_params;
+
+    for (const auto& enabled :
+         base::FeatureList::SplitFeatureListString(enable_features)) {
+      if (!base::FeatureList::ParseEnableFeatureString(
+              enabled, &feature, &study, &group, &feature_params)) {
+        // TODO(b/407734134): Register UMA for non-parsable features.
+        LOG(ERROR) << "Unable to parse enabled feature: " << enabled;
+        continue;
+      }
+      if (!feature.empty() && feature[0] == '*') {
+        override_map[feature] =
+            h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+      }
+      override_map[feature] =
+          h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+    }
+    for (const auto& disabled :
+         base::FeatureList::SplitFeatureListString(disable_features)) {
+      if (!base::FeatureList::ParseEnableFeatureString(
+              disabled, &feature, &study, &group, &feature_params)) {
+        // TODO(b/407734134): Register UMA for non-parsable features.
+        LOG(ERROR) << "Unable to parse disabled feature: " << disabled;
+        continue;
+      }
+      override_map[feature] =
+          h5vcc_experiments::mojom::OverrideState::OVERRIDE_DISABLE_FEATURE;
+    }
+  }
+  auto it = override_map.find(feature_name);
+  if (it != override_map.end()) {
+    return it->second;
+  }
+  return h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+}
+
+std::string GetFeatureParamInternal(const std::string& feature_param_name) {
+  return base::GetFieldTrialParamValue(cobalt::kCobaltExperimentName,
+                                       feature_param_name);
+}
+
+}  // namespace
+
+// TODO (b/395126160): refactor mojom implementation on Android
```

Comment:
I'm not sure this TODO applies to this... it mentions moving the implementation into the Java layer, which I don't think we'd ever do for H5vccExperiments. It needs to be in C++ such that it can access FeatureList in a platform-agnostic way.